ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/signing/hashicorp/HashicorpSignerHelper.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.signing.hashicorp;
+
+import tech.pegasys.ethsigner.core.signing.CredentialTransactionSigner;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.JsonObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.web3j.crypto.Credentials;
+
+public class HashicorpSignerHelper {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final Vertx vertx = Vertx.vertx();
```

Comment:
can't we just the existing vertx instance instead creating another?

Reply:
The other vertx is private in Runner. It is created later than this one. We could create one and then pass it into the runner ... 
What do you think?

Reply:
👍

Reply:
done